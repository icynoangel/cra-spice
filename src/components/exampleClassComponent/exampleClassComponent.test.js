import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import intl, { WrapIntlProvider } from '../../testsHelper';
import ExampleClassComponent from './ExampleClassComponent';

describe('<ExampleClassComponent />', function () {
  this.getComponent = () => {
    return (
      <WrapIntlProvider>
        <ExampleClassComponent
          onClick={this.handleClick}
          initialCounter={this.initialCounter}
        />
      </WrapIntlProvider>
    );
  };

  beforeEach(() => {
    this.initialCounter = 10;
    this.handleClick = jest.fn();
  });

  it('should render <ExampleClassComponent />', () => {
    const { container } = render(this.getComponent());
    expect(container).toMatchSnapshot();
  });

  it('should increase counter when button is clicked', () => {
    render(this.getComponent());
    userEvent.click(screen.getByRole('button'));
    expect(this.handleClick).toHaveBeenCalled();
    expect(screen.getByTestId('qa-counter-value').textContent).toEqual(
      `${intl.formatMessage({ id: 'counter' })}: 11`
    );
  });
});
