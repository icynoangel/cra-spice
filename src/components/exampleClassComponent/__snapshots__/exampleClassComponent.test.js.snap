// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExampleClassComponent /> should increase counter when button is clicked 1`] = `
<div
  className="example-class-component"
>
  <span
    className="example-class-component__counter"
  >
    <FormattedMessage
      defaultMessage="Counter"
      id="counter"
      values={Object {}}
    />
    : 
    11
  </span>
  <button
    className="example-class-component__button"
    onClick={[Function]}
  >
    <FormattedMessage
      defaultMessage="Click Me"
      id="clickMe"
      values={Object {}}
    />
  </button>
</div>
`;

exports[`<ExampleClassComponent /> should render <ExampleClassComponent /> 1`] = `
<div
  className="example-class-component"
>
  <span
    className="example-class-component__counter"
  >
    <FormattedMessage
      defaultMessage="Counter"
      id="counter"
      values={Object {}}
    />
    : 
    10
  </span>
  <button
    className="example-class-component__button"
    onClick={[Function]}
  >
    <FormattedMessage
      defaultMessage="Click Me"
      id="clickMe"
      values={Object {}}
    />
  </button>
</div>
`;
